### PR TITLE
165 rework card component resolve overflow icon and image boolean issues implement new structure

### PR DESCRIPTION
**Pull request by tishoyanchev**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

- Created ifx-card-image component
- fixed the ifx-link icon 
- removed all the slots for the content components, and replaced them with a single default one. The img still needs a dedicated slot, though.
- fixed ifx-link overflow bug
- removed Props used as internal States
- Added a React component for the Card

<details>
  <summary>📦 Published PR as canary version: <code>19.0.0--canary.201.d871b0728b738ebdc7d056496be26d29e5fc1b43.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @infineon/infineon-design-system-stencil@19.0.0--canary.201.d871b0728b738ebdc7d056496be26d29e5fc1b43.0
  # or 
  yarn add @infineon/infineon-design-system-stencil@19.0.0--canary.201.d871b0728b738ebdc7d056496be26d29e5fc1b43.0
  ```
</details>

### CODE REVIEW

**Review by verena-ifx — COMMENTED** [review]
Can you please make sure to align with the existing components in the example apps? Here, every component is in its own folder, but the card is just created as a file without a folder. Same goes for the naming and headlines in the example apps. 
We will work on an alignment between the different frameworks in the end, but for now, let's make sure that within one example app, everything is aligned already.

File: examples/wrapper-components/react-typescript/src/App.tsx [hunk 1]
```diff
@@ -44,13 +48,17 @@ function App() {
...
+         <h2>Tabs</h2>
+         <Tabs />
+         <br/>
 
-      <h2>Tabs</h2>
-      <Tabs />
-      <br/>
+         <h2>Tabs</h2>
+         <Card />
+         <br/>
```
Comment (verena-ifx) [inline examples/wrapper-components/react-typescript/src/App.tsx:61]: should be Card, that headline is what i meant

Comment (tishoyanchev) [reply examples/wrapper-components/react-typescript/src/App.tsx:61]: I know. I already noticed it, and changed it :D

**Review by verena-ifx — COMMENTED** [review]
you need to add the tags: autodocs, otherwise there will be no story on storybook
Update: Will do it in my review

**Comment by verena-ifx** [comment]
Added the direction to the argsTable in the Card story so that the user can configure the card direction and then get the according code snippet right away. Besides that, looks good! You can merge it

**Pull request merged**